robocup_ws/src/Planner/src/team_0_master.py

```diff
@@ -3,19 +3,35 @@
 from team_master import TeamMasterServer, TeamMaster
 from goalkeeper_controller import Team1GoalkeeperController
 from stay_in_place_controller import NullController
+from decision_tree import DecisionTree
+from player_controller import PlayerController
 
 
 class TeamMaster0(TeamMaster):
     def __init__(self):
         team_id = 0
         super().__init__(team_id)
-        self.goalkeeper_logic = Team1GoalkeeperController(0)
+        self.goalkeeper_logic = Team1GoalkeeperController(team_id)
         self.striker_left_logic = NullController()
         self.striker_right_logic = NullController()
         self.defence_left_logic = NullController()
         self.defence_right_logic = NullController()
+        self.players = {self.defence_left_logic: "DEFEND",
+                        self.defence_right_logic: "DEFEND",
+                        self.striker_left_logic: "ATTACK",
+                        self.striker_right_logic:"ATTACK" }
+        self.tree = None
         self.players_logic_was_updated = True
 
+    def plan(self):
+        super(TeamMaster0, self).plan()
+        player_id = 0
+        for player, mode in self.players.items():
+            player = PlayerController(player_id)
```

Comment:
why not to in ```__init__()```
```
self.striker_left_logic = PlayerController(player_id)
self.striker_left_tree = DecisionTree(pc = self.striker_left_logic)
```

and here:
```
self.actions[self.striker_left_id] = self.striker_left_tree.plan(self.team_position,self.opponents_position, player_id, mode)
...
```
this would significantly speed the flow and save garbage collector plenty of work